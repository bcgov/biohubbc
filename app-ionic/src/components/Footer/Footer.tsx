import React from 'react';
import { Link, Toolbar } from '@material-ui/core';

function Footer() {
  return (
    <div className="App-footer">
      <Toolbar>
        <Link href="http://www.gov.bc.ca/gov/content/home/disclaimer">Disclaimer</Link>
        <Link href="http://www.gov.bc.ca/gov/content/home/privacy">Privacy</Link>
        <Link href="http://www.gov.bc.ca/gov/content/home/accessible-government">Accessibility</Link>
        <Link href="http://www.gov.bc.ca/gov/content/home/copyright">Copyright</Link>
      </Toolbar>
    </div>
  );
}

export default Footer;
