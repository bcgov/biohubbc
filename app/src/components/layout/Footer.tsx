import { Box, createStyles, Link, makeStyles, Theme, Typography } from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    box: {
      backgroundColor: theme.palette.primary.main
    },
    typography: {
      color: theme.palette.primary.contrastText
    },
    link: {
      color: 'inherit',
      padding: theme.spacing(0, 2, 0)
    }
  })
);

const Footer: React.FC = () => {
  const classes = useStyles();

  return (
    <Box p={2} className={classes.box}>
      <Typography className={classes.typography}>
        <Link className={classes.link} href="http://www.gov.bc.ca/gov/content/home/disclaimer">
          Disclaimer
        </Link>
        <Link className={classes.link} href="http://www.gov.bc.ca/gov/content/home/privacy">
          Privacy
        </Link>
        <Link className={classes.link} href="http://www.gov.bc.ca/gov/content/home/accessible-government">
          Accessibility
        </Link>
        <Link className={classes.link} href="http://www.gov.bc.ca/gov/content/home/copyright">
          Copyright
        </Link>
      </Typography>
    </Box>
  );
};

export default Footer;
