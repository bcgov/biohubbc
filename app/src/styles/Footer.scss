@import 'colours.scss';

$footerNavSpacing: 1rem;

.app-footer {
  background-color: $BCgovBlue;
  border-top: 2px solid $BCgovGold;
  color: $appFootercolor;
}

.app-footer ul {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  margin: 0;
  padding: 0;
  list-style: none;
}

.app-footer li + li {
  margin-left: $footerNavSpacing;
  padding-left: $footerNavSpacing;
  border-left: 1px solid #4b5e7e;
}

.app-footer a {
  color: #fff;
  text-decoration: none;
  font-size: 0.875rem;

  &:hover {
    color: #fff;
    text-decoration: underline;
  }

  // &:focus {
  //   outline: 4px solid #3B99FC;
  //   outline-offset: 1px;
  // }
}

.app-footer nav {
  align-items: center;
}

.app-footer-toolbar {
  min-height: 46px !important;
}
