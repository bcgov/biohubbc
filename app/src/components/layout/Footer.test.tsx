import { render } from '@testing-library/react';
import React from 'react';
import Footer from './Footer';

describe('Footer', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Footer />);

    expect(asFragment()).toMatchSnapshot();
  });
});
