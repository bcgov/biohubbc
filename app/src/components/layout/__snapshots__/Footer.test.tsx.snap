// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<DocumentFragment>
  <footer
    class="makeStyles-appFooter-1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl"
    >
      <div
        aria-label="Footer"
        class="MuiToolbar-root MuiToolbar-regular makeStyles-appFooterToolbar-2"
        role="navigation"
      >
        <ul>
          <li>
            <a
              href="https://www.gov.bc.ca/gov/content/home/disclaimer"
            >
              Disclaimer
            </a>
          </li>
          <li>
            <a
              href="https://www.gov.bc.ca/gov/content/home/privacy"
            >
              Privacy
            </a>
          </li>
          <li>
            <a
              href="https://www.gov.bc.ca/gov/content/home/accessible-government"
            >
              Accessibility
            </a>
          </li>
          <li>
            <a
              href="https://www.gov.bc.ca/gov/content/home/copyright"
            >
              Copyright
            </a>
          </li>
        </ul>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
